plaid/src/data/okta/mod.rs

```diff
@@ -1,59 +1,107 @@
+    /// If no value is provided here, we will default to 1 milliseconds between calls
+    sleep_duration: Option<u64>,
     #[serde(default)]
     pub logbacks_allowed: LogbacksAllowed,
 }
 
+/// Custom parser for limit. Returns an error if a limit = 0 or limit > 1000 is given
+fn parse_limit<'de, D>(deserializer: D) -> Result<Option<u16>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let limit = Option::<u16>::deserialize(deserializer)?;
+
+    if let Some(limit) = limit {
+        match limit {
+            1..=1000 => Ok(Some(limit)),
+            0 => Err(serde::de::Error::custom(
+                "Invalid limit value provided. Minimum limit is 1",
+            )),
+            _ => Err(serde::de::Error::custom(
+                "Invalid limit value provided. Maximum limit is 1000",
+            )),
+        }
+    } else {
+        Ok(None)
+    }
+}
+
 pub struct Okta {
+    /// A `reqwest` client to send API calls with
     client: Client,
+    /// Contains domain and token
     config: OktaConfig,
+    /// Filters the lower time bound of the log events published property for bounded queries or persistence time for polling queries
     since: OffsetDateTime,
+    /// Sending channel used to send logs into the execution system
     logger: Sender<Message>,
 }
 
 /// We try not to parse anything complicated since our job is just
 /// to pass it on.
+/// See https://developer.okta.com/docs/reference/api/system-log/#logevent-object for full docs
 #[derive(Deserialize, Serialize)]
 struct OktaLog {
+    /// Timestamp when the event is published
     published: String,
+    /// Describes the entity that performs an action
     actor: Value,
+    /// The client that requests an action
     client: Value,
+    /// Type of device that the client operates from (for example, Computer)
     device: Value,
+    /// The authentication data of an action
     #[serde(rename = "authenticationContext")]
     authentication_context: Value,
+    /// The display message for an event
     #[serde(rename = "displayMessage")]
     display_message: Value,
+    /// Type of event that is published
     #[serde(rename = "eventType")]
     event_type: Value,
+    /// The outcome of an action
     outcome: Value,
+    /// The security data of an action
     #[serde(rename = "securityContext")]
     security_context: Value,
+    /// Indicates how severe the event is: DEBUG, INFO, WARN, ERROR
     severity: Value,
+    /// The debug request data of an action
     #[serde(rename = "debugContext")]
     debug_context: Value,
+    /// Associated Events API Action
     #[serde(rename = "legacyEventType")]
     legacy_event_type: Value,
+    /// The transaction details of an action
     transaction: Value,
+    /// Unique identifier for an individual event
     uuid: Value,
+    /// Versioning indicator
     version: Value,
+    /// The request that initiates an action
     request: Value,
+    /// Zero or more targets of an action
```

Comment:
Does this struct generate a whole bunch of unused warnings? If we don't need to parse it maybe we should just hide it behind a `Value` and pull out the published field manually